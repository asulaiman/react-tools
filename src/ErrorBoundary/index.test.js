/* eslint-env jest */

import React, { Component } from 'react'
import renderer from 'react-test-renderer'
import { ErrorBoundary } from './'

console.error = jest.fn()

class ErrorView extends Component {
  throwErrorInRender() {
    throw new Error('YOLO')
  }

  render() {
    return <div>{this.throwErrorInRender()}</div>
  }
}

describe('Error Boundary', () => {
  it('matches snapshot', () => {
    let component = renderer.create(
      <ErrorBoundary>
        <ErrorView />
      </ErrorBoundary>
    )

    expect(component.toJSON()).toMatchSnapshot()
    expect(console.error).toHaveBeenCalled()
  })
})
