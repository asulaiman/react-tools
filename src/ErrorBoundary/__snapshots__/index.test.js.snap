// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary matches snapshot 1`] = `
<h1>
  Something has gone wrong.
</h1>
`;
